fastlane/Fastfile

```diff
@@ -1,27 +1,207 @@
+        build_and_upload_app_store(options)
+    end
+
+    def build_and_upload_app_store(options)
+        options[:build_number] = generate_build_number
+        prepare_for_release(options)
+        if options[:run_security_tests]
+            security_plans(options)
+        end
+        build_for_release(options)
+        upload_app_store(options)
+        upload_s3(options)
+        if options[:upload_dsyms_to_app_center]
+            upload_dsyms_to_app_center_for_appstore_build(options)
+        end
+    end
+
+    def fetch_latest_configuration_repo 
+        path = '../wire-ios-build-assets'
+        `rm -rf #{path}`
+        if is_running_on_ci 
+            Git.clone(
+                "https://#{ENV['WIRE_INTERNAL_GITHUB_USER']}:#{ENV['WIRE_INTERNAL_GITHUB_TOKEN']}@github.com/wireapp/wire-ios-build-assets.git",
+                path
+            )
+        else
+            Git.clone(
+                "git@github.com:wireapp/wire-ios-build-assets.git",
+                path
+            )
+        end
+
+    end
+
+    def download_distribution_certificate_and_profiles(build)
+        if is_running_on_ci 
+            match(
+                git_url: "https://#{ENV['WIRE_INTERNAL_GITHUB_USER']}:#{ENV['WIRE_INTERNAL_GITHUB_TOKEN']}@github.com/wireapp/wire-ios-certificates.git",
+                readonly: true,
+                type: "appstore",
+                skip_provisioning_profiles: true,
+                app_identifier: []
+            )
+        else
+            match(
+                readonly: true,
+                type: "appstore",
+                skip_provisioning_profiles: true,
+                app_identifier: []
+            )
+        end
+
+        app_bundle_id = get_xcconfig_value(
+            path: './wire-ios/Configuration/Release.xcconfig',
+            name: 'WIRE_BUNDLE_ID'
+        )
+        puts "downloading provisioning profiles for bundle id: #{app_bundle_id}"
+        should_sigh_adhoc = !build.appstore_build && !build.rc_build && !build.beta_build
+        sigh(
+            adhoc: should_sigh_adhoc,
+            readonly: true,
+            api_key_path: appstoreconnect_api_key_file_path,
+            app_identifier: app_bundle_id
+        )
+        sigh(
+            adhoc: should_sigh_adhoc,
+            readonly: true,
+            api_key_path: appstoreconnect_api_key_file_path,
+            app_identifier: "#{app_bundle_id}.extension-share"
+        )
+        sigh(
+            adhoc: should_sigh_adhoc,
+            readonly: true,
+            api_key_path: appstoreconnect_api_key_file_path,
+            app_identifier: "#{app_bundle_id}.extension-notification"
         )
     end
-    
 
-    desc "Fetch dependencies"
-    lane :prepare do |options|
+    def generate_build_number
+        build_number = `curl -H "Accept: text/plain" -X POST -d '' "https://incrementator.appspot.com/api/v1/code/ORUHILeMbXB4D5Z5conFzYoYpQPjVnyUA4pu56X2MHVJSFxM/increment"`
```

Comment:
is it safely hosted ? is there a way to host it?

Reply:
What do you mean by any unsafeness here?

Reply:
I mean who owns this script? is it a service? what happens if it goes down.

Reply:
From my experience, it never went down and I used it in several apps... It is available to host a similar service on our own but I'm wondering does it make sense to implement and possibly further maintain such software when there is such a reliable solution (may not look as reliable 😅  but it is...)  
Let me know what you think about it :)

Reply:
it's just that I want to know more info. I don't know who created it, maintains it, who can I reach if it has some issue